src/test/groovy/WithGoEnvUnixStepTests.groovy

```diff
@@ -122,17 +122,19 @@ void testOSArg() throws Exception {
   @Test
   void testDefaultGoVersion() throws Exception {
     helper.registerAllowedMethod('nodeOS', [], { "linux" })
+    def scriptDV = loadScript('vars/goDefaultVersion.groovy')
+    def version = scriptDV.defaultVersion()
```

Comment:
Uhm, when I used `step`.method in the past, I used to use a mock class and then use the `helper.property` method, see https://github.com/elastic/apm-pipeline-library/blob/85bc3987206c6aad98744a576c6e61e8692374be/src/test/groovy/ApmBasePipelineTest.groovy#L82-L86

What do you think?

Reply:
the use of this step is to do not have to modify the test every time we change the default Go version, if we use a mock we have to change the version in the step and in the mock.

Reply:
Maybe `defaultVersion` should be an unique step, renamed to `staticGoDefaultVersion` and then it will be easy to be mocked in the context of this particular class with:

```
    helper.registerAllowedMethod('staticGoDefaultVersion', [], { "1.15.5" })
    if(binding.getVariable("PATH") == "WS/bin:WS/.gvm/versions/go1.15.5.linux.amd64/bin:/foo/bin"
       && binding.getVariable("GOROOT") == "WS/.gvm/versions/go1.15.5.linux.amd64"
...
```

Reply:
I am not sure you can mock a method from the same class in that way. I will check.

Reply:
I realized that I can mock the function directly